### PR TITLE
Validate image format before resize in Resizetizer

### LINKED ISSUE #6166: NullReferenceException when Resources/Images contains non-image
### Description

In my test project I created an `attributions.txt` file in the `Resources/Images/` folder. 

```
Error    MAUI0000    System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Maui.Resizetizer.SkiaSharpBitmapTools.GetOriginalSize() in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\SkiaSharpBitmapTools.cs:line 29
   at Microsoft.Maui.Resizetizer.SkiaSharpTools.Resize(DpiPath dpi, String destination) in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\SkiaSharpTools.cs:line 47
   at Microsoft.Maui.Resizetizer.Resizer.Resize(DpiPath dpi, String inputsFile) in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\Resizer.cs:line 101
   at Microsoft.Maui.Resizetizer.ResizetizeImages.ProcessImageResize(ResizeImageInfo img, DpiPath[] dpis, ConcurrentBag`1 resizedImages) in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\ResizetizeImages.cs:line 174
   at Microsoft.Maui.Resizetizer.ResizetizeImages.<>c__DisplayClass27_0.<ExecuteAsync>b__0(ResizeImageInfo img) in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\ResizetizeImages.cs:line 90
   at Microsoft.Maui.Resizetizer.AsyncTaskExtensions.<>c__DisplayClass0_0`1.<ParallelForEach>b__0(TSource s) in D:\a\_work\1\s\src\SingleProject\Resizetizer\src\AsyncTaskExtensions.cs:line 21    ERecept-MAUI    C:\Program Files\dotnet\packs\Microsoft.Maui.Resizetizer.Sdk\6.0.200-preview.14.5099\targets\Microsoft.Maui.Resizetizer.targets    475   
```

I found that VS automatically adds the `MauiImage` build action to everything it adds to this folder. Including the `.txt`.

Probably related SO post: <https://stackoverflow.com/questions/70747813>

The error message should be more specific (file that caused it?) rather than a NRE.

### Steps to Reproduce

Put anything that is not an image into `Resources/Images/` and mark it as a `MauiImage` (build action)

### Version with bug

Preview 14

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11

### Did you find any workaround?

Check that all contents of the `Resources/Images/` folder are "image-like" files

### Relevant log output

A blank warning pointing to a targets file
```shell
Warning		MAUI0000	ERecept-MAUI	C:\Program Files\dotnet\packs\Microsoft.Maui.Resizetizer.Sdk\6.0.200-preview.14.5099\targets\Microsoft.Maui.Resizetizer.targets	475	
```
and the error log above.

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Validate image format before resize in Resintetizer.

### Issues Fixed

Fixes #6166

### CODE REVIEW

File: src/SingleProject/Resizetizer/src/ResizeImageInfo.cs [hunk 1]
```diff
@@ -61,11 +63,15 @@ public static List<ResizeImageInfo> Parse(IEnumerable<ITaskItem> images)
...
 			{
 				var info = new ResizeImageInfo();
-
+								
 				var fileInfo = new FileInfo(image.GetMetadata("FullPath"));
+
 				if (!fileInfo.Exists)
 					throw new FileNotFoundException("Unable to find background file: " + fileInfo.FullName, fileInfo.FullName);
 
+				if (!ValidImageExtensions.Contains(fileInfo.Extension.ToUpperInvariant()))
```
Comment (jsuarezruiz) [inline src/SingleProject/Resizetizer/src/ResizeImageInfo.cs:72]: Weak validation? We could try to open the image, getting the byte array and verifying the extension information to be more precise.

**Comment by mattleibow** [comment]
Maybe we need to edit the targets file to not include things sbf only include specific extensions

File: src/SingleProject/Resizetizer/src/ResizeImageInfo.cs [hunk 2]
```diff
@@ -61,11 +63,15 @@ public static List<ResizeImageInfo> Parse(IEnumerable<ITaskItem> images)
...
+					throw new FileNotFoundException("The file is not a valid image format: " + fileInfo.FullName, fileInfo.FullName);
```
Comment (dersia) [inline src/SingleProject/Resizetizer/src/ResizeImageInfo.cs:73]: Is FileNotFound really the best exception?
File is found, it more of a format exception, isn't it?

**Comment by jsuarezruiz** [comment]
To help focus and move faster, we're going to have fewer open PRs and focus on the ones we'll be working on in each release. This PR is not included in the next release so I close it for now. I will reopen it and adapt to changes later.

**Pull request closed without merging**